### ISSUE TITLE
Invisible proxy icon

### ISSUE THREAD

**Issue by ilyagr** [macvim-dev/macvim#1503]
### Steps to reproduce

1. Run `mvim file.md` without MacVim being open
2. Observe the lack of the proxy icon

<img width="688" alt="image" src="https://github.com/user-attachments/assets/188ae41f-17e3-4824-b6fe-572af753d1d8">

That's the problem, but here's a workaround.

3. Go to appearance settings

<img width="595" alt="image" src="https://github.com/user-attachments/assets/ea1a5d4e-6b69-4db8-bfef-73a9f5e6f2b3">

4. Check "Hidden title bar" and uncheck it again
5. The proxy icon is visible now.

<img width="688" alt="image" src="https://github.com/user-attachments/assets/c8929962-3730-4edd-b63c-d8af84853023">

 

### Expected behaviour

The proxy icon should be visible regardless.

### Version of Vim and architecture

MacVim r180, Vim 9.1.727, ARM, on 

### Environment

Sonoma 14.7

I'm using the dark theme for it.

### How MacVim was installed

From GitHub

### Issue has been tested with given configuration

- [ ] by running MacVim.app from GUI macOS interface
- [X] by running vim/gvim/etc installed by MacVim
- [ ] by running other versions of vim (e.g. /usr/bin/vim)

### Issue has been tested with no configuration

- [X] by running `mvim --clean` (or `gvim`, supplied by MacVim distribution)
- [ ] by running `vim --clean` (in terminal, supplied by MacVim distribution)
- [ ] by running `vim --clean` (in terminal, other suppliers, e.g. /usr/bin/vim)

### Other conditions

- [ ] The both Homebrew packages "vim" and "macvim" are installed

**Comment by ychin** [comment]
This was actually done intentionally when I implemented #1038. It isn't really a bug as I just thought this looks cleaner, but in retrospect providing an option to show file name or not might have been the easier / cleaner solution.

Was the motivation for this issue post entirely for the ripdrag linked issue? I was curious what you use the icon for.

**Comment by ilyagr** [comment]
I would currently mainly use it for that, yes. I may or may not open the file only to drag it.

It feels like a bug that toggling the "hide window bar" option changes the look. 

By default, Mac OS Sonoma hides the icon until you click on the window title. I'm not sure how easy that is, but ideally you'd follow that behavior and respect the "show window icons" system option for people who want to avoid that click.

**Comment by ychin** [comment]
Yeah I think having a separate option properly makes more sense. Sometimes trying to be too smart is detrimental to having a consistent setting. I'll put this to backlog.

> By default, Mac OS Sonoma hides the icon until you click on the window title. I'm not sure how easy that is, but ideally you'd follow that behavior and respect the "show window icons" system option for people who want to avoid that click.

I think this only applies to the combined toolbar mode. For example, Finder and Preview does that, but TextEdit doesn't.

In fact, MacVim also does that if you use the toolbar as well (`:set go+=T`). Either way it wouldn't affect our decision here.
